bears/uploadTool.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import glob
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+# inserting path here so the following imports work wherever you run the script
+# in coala-bears
+sys.path.insert(0, '..')
+
+from coalib.collecting.Importers import iimport_objects
+from string import Template
+
+
+def create_directory(folder_name):
+    """
+    Checks if the folder exists, and creates it if it doesn't.
+
+    :param name: The directory name.
+    """
+    if not os.path.exists(folder_name):
+        os.mkdir(folder_name)
+
+
+def touch(file_name):
+    """
+    Creates an empty file.
+
+    :param file_name: Name of the file.
+    """
+    if not os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def main():
+    create_directory('upload')
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = next(iimport_objects(file_name, attributes='kind',
+                                           local=True))
+        create_directory(os.path.join('upload', bear_name))
+        create_directory(os.path.join('upload', bear_name, bear_name))
+        touch(os.path.join('upload', bear_name, bear_name,
+                           '__init__.py'))
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        with open('setup.py.in') as fl:
+            template = Template(fl.read())
+
+        NAME = bear_name
+        VERSION = '1.0.30'  # testing purposes, not sure yet how we hold it
+        AUTHORS = str(bear_object.AUTHORS)
+        AUTHORS_EMAILS = str(bear_object.AUTHORS_EMAILS)
+        MAINTAINERS = str(bear_object.maintainers)
+        MAINTAINERS_EMAILS = str(bear_object.maintainers_emails)
+        PLATFORMS = str(bear_object.PLATFORMS)
+        LICENSE = str(bear_object.LICENSE)
+        LONG_DESCRIPTION = str(bear_object.__doc__)
+
+        processed_template = template.safe_substitute(**locals())
+
+        with open(os.path.join('upload', bear_name, 'setup.py'), "w") as setup:
+            setup.write(processed_template)
+
+            os.chdir(os.path.join('upload', bear_name))
+            parser = argparse.ArgumentParser(description='Choose to upload'
+                                             'the bears or not')
+            parser.add_argument('-u', '--upload', help='Upload the bears',
+                                action='store_true')
+            args = parser.parse_args()
```

Comment:
would you mind doing argparsing in the beginning please? parsing args might show the help and just exit out directly e.g.